File: src/main/java/com/eova/core/role/RoleIntercept.java
/**
 * Copyright (c) 2019 EOVA.CN. All rights reserved.
 * <p>
 * Licensed to: 卢中强(495993331@qq.com)
 * Licensed under the EPPL license: http://eova.cn/eppl.txt
 * Software copyright registration number:2018SR1012969
 * For authorization, please contact: admin@eova.cn
 */
package com.eova.core.role;

import com.eova.aop.AopContext;
import com.eova.aop.MetaObjectIntercept;

public class RoleIntercept extends MetaObjectIntercept {

    @Override
    public String addBefore(AopContext ac) throws Exception {
        Integer lv = ac.record.getInt("lv");
        Integer roleLv = ac.user.getRole().getInt("lv");
        if (lv <= roleLv) {
            return error("权限级别必须大于：" + roleLv);
        }
        return null;
    }

    @Override
    public String updateBefore(AopContext ac) throws Exception {
        return addBefore(ac);
    }


}

File: src/main/kotlin/cn/doit/natp/intercept/RoleIntercept.kt
@file:Suppress("unused")

package cn.doit.natp.intercept

import cn.doit.natp.common.BaseMetaObjectIntercept
import com.eova.aop.AopContext
import com.eova.aop.MetaObjectIntercept

class RoleIntercept : BaseMetaObjectIntercept() {

    override fun addBefore(ac: AopContext): String? {
        val lv = ac.record.getInt("lv")
        val roleLv = ac.user.getRole().getInt("lv")
        return if (lv <= roleLv) MetaObjectIntercept.error("权限级别必须大于：" + roleLv!!) else null
    }

    override fun updateBefore(ac: AopContext): String? {
        return this.addBefore(ac)
    }

}